### ISSUE TITLE
Do not change volumes when activating "standard" test flights

### ISSUE THREAD

**Issue by BenjaminPelletier** [interuss/monitoring#461]
**Is your feature request related to a problem? Please describe.**
Currently, flight intents for F3548 testing have a fixed duration relative to a variable starting point.  So, when the intent is evaluated at activation time, its end time will shift slightly forward of the end time it had when it was just planned.  While this is not prohibited behavior, a common use case would likely be for a user to activate the exact flight plan they had planned with no volumes changes apart from the start time perhaps being moved forward to "now".

**Describe the solution you'd like**
Change standard injected flight intents to have a fixed end time when transitioning from Planned to Activated.

**Describe alternatives you've considered**
We could keep things as-is and force testing different portions of the system (changing volumes at the same time as activating the flight), but in the interest of keeping simple tests simple, I think it would be better to change only one thing at a time.  We can test changing volumes at the same time as activating the flight in a separate test dedicated to exercising that particular functionality.

**Comment by Shastick** [comment]
An [example](https://github.com/interuss/monitoring/blob/5ab499fd8076714cc272a1a2824955f3c1acd31c/monitoring/uss_qualifier/scenarios/astm/utm/subscription_notifications/receive_notifications_for_awareness/receive_notifications_for_awareness.py#L171) illustrating this:

```
flight_1_planned = self.flight_1_planned.resolve(times)
flight_1_activated = self.flight_1_activated.resolve(times)
```

(with `times` being a `Dict[TimeDuringTest, Time]`)

Solving this could be done by:
 - changing `resolve` so that we let it know about when a previous call to `resolve` was made
 - changing `times` to reach the same effect
 - updating the flight_intent configuration to allow controlling what happens on a state change 
 - having a separate method next to `resolve` that takes an existing plan and reuses its extends, or pass the previous plan as an optional argument to `resolve`
 
The last option seems to be the easiest to implement, but it forces test designers to pass the correct parameters: an alternative where the example code above would "just work" might be desirable.

**Comment by BenjaminPelletier** [comment]
I'm not sure we need a code change, and it would probably be tricky to accomplish this in code because we wouldn't want to change the contract/behavior expected for time specifications.  I was expecting something like changing "time of evaluation TO time of evaluation + 5 minutes" to "latest of time of evaluation or start of test scenario TO start of test scenario + 10 minutes".  Although it has the advantage of being very simple, the former will creep forward each time it is evaluated while the latter's end time will not creep forward.

**Comment by Shastick** [comment]
I see.

> I'm not sure we need a code change.

By this, do you mean, no code change in _scenarios_, but possibly in the geotemporal library, or no code change _at all_, and only update the configuration?

>  I was expecting something like changing "time of evaluation TO time of evaluation + 5 minutes" to "latest of time of evaluation or start of test scenario TO start of test scenario + 10 minutes"

(I'm not entirely up to date with the configuration possibilities of flight intents so don't know if the above is entirely doable through the configuration)

**Comment by BenjaminPelletier** [comment]
I don't think we would need any changes in the geotemporal library, but changes might be needed in scenarios if they don't use the standard FlightInfoTemplates yet.  FlightInfoTemplates allow interesting time specifications and I think using the right specification should resolve this issue.

The problem with the code in the link is that it evaluates the templates too early -- TimeOfEvaluation is supposed to be as close to "now" as we can reasonably get in the test.  If someone specified their flight info such that flight 1 planned was supposed to be now until 10 minutes from now and then flight 1 activated was supposed to be now until 10 minutes from now, it's true this issue would not arise there, but that's because TimeOfEvaluation is not updated to "effectively now" for flight 1 activated.  Once TimeOfEvaluation is updated just prior to evaluating the template for flight 1 activated just prior to when it is used, then TimeOfEvaluation will be a little later for flight 1 activated than it was for flight 1 planned and the issue will be observed.

**Comment by Shastick** [comment]
Dropping this for now in favor of issues with a higher priority.
